app/helpers/discovered_hosts_helper.rb

```diff
@@ -12,27 +12,22 @@ def disc_report_column(record)
   end
 
   def discovered_hosts_title_actions(host)
-    actions = [[_('Provision'),  hash_for_edit_discovered_host_path(:id => host)]]
-    actions <<  [_('Auto Provision'), hash_for_auto_provision_discovered_host_path(:id => host), :method => :post]
-    actions <<  [_('Refresh facts') ,hash_for_refresh_facts_discovered_host_path(:id => host)]
-    actions <<  [_('Reboot') ,hash_for_reboot_discovered_host_path(:id => host), :method => :put]
+    actions = [[_('Provision'), hash_for_edit_discovered_host_path(:id => host).merge(:auth_object => host, :permission => :provision_discovered_hosts)]]
+    actions <<  [_('Auto Provision'), hash_for_auto_provision_discovered_host_path(:id => host).merge(:auth_object => host, :permission => :auto_provision_discovered_hosts), :method => :post]
+    actions <<  [_('Refresh facts') ,hash_for_refresh_facts_discovered_host_path(:id => host).merge(:auth_object => host, :permission => :edit_discovered_hosts)]
+    actions <<  [_('Reboot') ,hash_for_reboot_discovered_host_path(:id => host).merge(:auth_object => host, :permission => :edit_discovered_hosts), :method => :put]
```

Comment:
Why adding these elements to the array instead of declaring it right away as an array containing all of them?
Unless you want to only add those that are permitted, as we do for example in https://github.com/theforeman/foreman/blob/develop/app/helpers/hosts_helper.rb#L148 and then you don't need to check the permissions again below.